runtime.go

```diff
@@ -591,7 +625,79 @@ func getWorkingDir(basedir string, codeuri string, checkWorkingDirExist bool) (s
 
 	// Windows uses \ as the path delimiter, but Docker requires / as the path delimiter.
 	dir = filepath.ToSlash(dir)
-
 	return dir, nil
 
 }
+
+// decompressArchive unzips a ZIP archive to a temporary directory and returns
+// the temporary directory name, or an error
+func decompressArchive(src string) (string, error) {
+
+	// Create a temporary directory just for this decompression (dirname: OS tmp directory + unix timestamp))
+	tmpdir := os.TempDir()
+
+	// By default on OSX, os.TempDir() returns a directory in /var/folders/.
+	// This sits outside the default Docker Shared Files directories, however
```

Comment:
So another change in this PR is to show Docker errors when they occur (previously they were not shown). For example:

```
2017/08/18 15:58:47 Successfully parsed samples/java/template.yml (version 2010-09-09)
2017/08/18 15:58:47 Connected to Docker 1.30
2017/08/18 15:58:47 Fetching lambci/lambda:java8 image for java8 runtime...
java8: Pulling from lambci/lambda
Digest: sha256:668979530863a155b52533947e782fe164908eda2d6c5741317511bfa036d8cf
Status: Image is up to date for lambci/lambda:java8
2017/08/18 15:58:49 Reading invoke payload from stdin (you can also pass it from file with --event)
2017/08/18 15:58:49 Invoking com.aws.codestar.projecttemplates.handler.HelloWorldHandler (java8)
2017/08/18 15:58:49 Decompressing HelloWorld-1.0.jar into runtime container...
2017/08/18 15:58:50 Could not invoke function: Error response from daemon: Mounts denied:
The path /var/folders/l1/s8wdtvds75n6x4tdkk0ryhf0c753jq/T/aws-sam-local-1503068329451744569
is not shared from OS X and is not known to Docker.
You can configure shared paths from Docker -> Preferences... -> File Sharing.
See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.
```